### DISCUSSION TITLE
bed temperature of first layer carrying on to the rest of the layers

### DISCUSSION THREAD

**Discussion by Studioscale** [OrcaSlicer/OrcaSlicer#1492] [General]
i set the slicer up so that the first layer would be 120 and the rest 110 but its printing the whole thing at 120. The extruder is doinf it correctly, reducing from 250 to 240 for the second layer on, but not the bed temp. Am I doing something wrong here?
![Orca temp](https://github.com/SoftFever/OrcaSlicer/assets/130306639/5e9e3365-a02d-4480-b4cd-2b8f6945ea20)
![orca temp screen](https://github.com/SoftFever/OrcaSlicer/assets/130306639/968a44bd-b2f8-4115-85ec-0a5c43dc6ba3)
[3DBenchy_ABS_43m38s.txt](https://github.com/SoftFever/OrcaSlicer/files/11990942/3DBenchy_ABS_43m38s.txt)

**Comment by Glenn1963** [comment]
So this is going to sound crazy. 

I started playing around with my Layer change G-code to create a workaround for your problem. While doing that I found I was able to generate g-code that did not include the necessary M140 command to change the bed temperature, just like in your sample g-code.

Somehow changing the Layer change G-code seems to turn off the Bed Temperature change.

1. Changed the Machine G-code for Layer Change
2. Sliced and Exported the G-code
3. No M140 command for Layer 2
4. Changed the High Temp Plate Other Layers value by 1 degree
5. Sliced and Exported the G-code
6. M140 command for Layer 2 now generated

So maybe something to try there. Starting with changing your _Other layers_ setting and Slicing again.

It's pretty consistent for me. If I modify the Layer change G-code then the bed temp isn't changed, until I change the Other layers temp.

If after all that you still cant get it to work, you can add this line to your Layer Change G-code
`{if layer_num == 1}M140 S[bed_temperature]; After Layer Change {endif}`
(Layer count starts at 0 internally)

**Comment by SoftFever** [comment]
@Glenn1963 
I can't reproduce it.
Can you share a project file?

**Reply by Studioscale** [reply]
I will try to reproduce it when I get home and post the project file.

**Reply by Glenn1963** [reply]
I can, but I can reproduce it on any. This cube, for example.
[cube.zip](https://github.com/SoftFever/OrcaSlicer/files/12217920/cube.zip)

It was slightly trickier for me today, I don't know if it's always been this way for me, but I'm on a 1.64 dev version now.

1. Clear the plate, add a new object
2. Note the current filament High Temp Plate / Other Layers setting value. Change it, but do not "Save current filament"
3. Slice and export g-code. The M140 with the new temperature should be there for Layer 2
4. Change the Machine g-code / Layer change g-code
5. Slice and export g-code. The M140 with the ORIGINAL High Temp Plate / Other Layers setting temperature should be there for Layer 2. NOT the new value you selected in step 2. I think this could be a separate issue, it seems like it is either resetting or reloading the filament after a machine change. 
6. Change the Machine g-code / Layer change g-code again.
7. Slice and export g-code. No M140
8. Profit

Here's what I'm seeing.
Step 3 g-code (The original layer 2 bed temperature value was 45, which I changed to 46):
 ```
;AFTER_LAYER_CHANGE
;0.5
M117 Layer 2/3
SET_VELOCITY_LIMIT ACCEL=8000 ACCEL_TO_DECEL=5000
M140 S46 ; set bed temperature
; printing object Cube.stl id:0 copy 0
G1 X8.939 Y8.939 Z1.2 F18000
```

Step 5 g-code  (the M190 is just my obvious throw-away layer change g-code):

```
;AFTER_LAYER_CHANGE
;0.5
M117 Layer 2/3
M190 S02
SET_VELOCITY_LIMIT ACCEL=8000 ACCEL_TO_DECEL=5000
M140 S45 ; set bed temperature
; printing object Cube.stl id:0 copy 0
G1 X8.939 Y8.939 Z1.2 F18000
```

Step 7 g-code:

```
;AFTER_LAYER_CHANGE
;0.5
M117 Layer 2/3
M190 S03
SET_VELOCITY_LIMIT ACCEL=8000 ACCEL_TO_DECEL=5000
; printing object Cube.stl id:0 copy 0
G1 X8.939 Y8.939 Z1.2 F18000
```

**Comment by jamesarthur** [comment]
This happens 100% of the time for me. It only ever sets the bed to whatever I set the first layer bed temp to. Klipper gcode.

**Comment by SoftFever** [comment]
Hey guys,

I need more info and and a sample project that can repro the issue to investigate it.

**Reply by Glenn1963** [reply]
[Project](https://github.com/SoftFever/OrcaSlicer/files/12217920/cube.zip)
(Not sure how Project files work, or if you can load my project that uses my own profiles)

Using this project on 1.6.6, the detailed steps I listed above are not 100% accurate. In that if you simply:

1. Load this project
2. Slice: You will get the M140
3. Change the Machine g-code / Layer change g-code
4. Slice: You will no longer get an M140 

I would say it's almost definitely settings-related somewhere, since the settings I currently use in 1.6.6 DO NOT generate this problem, but this project does.
Ok, well, I tried again using my current settings but a smaller layer size, and I can again reproduce this. Basically make sure there are more than 2 layers I think. Another edit: Oof, nope. I switch between 2 presets, one can produce this issue, one doesn't. I am still trying to compare them to find out why.

**Reply by Glenn1963** [reply]
As far as I can tell it may have something to do with 
Label Objects
Exclude Objects
Verbose G-Code (maybe)

If I have all 3 off I can get it to fail to generate an M140 _after_ I change Machine g-code / Layer change g-code

**Reply by jamesarthur** [reply]
I'm not able to play with my settings right now but they are...
* Verbose G-Code - OFF
* Label Objects - ON
* Exclude Objects - ON

This is the config that fails to generate the M140 after the 1st layer.

**Comment by agravelot** [comment]
Had this issue when using two filaments project, one filament per plate.

**Comment by 35C0lDFuSi0n** [comment]
I started having this problem when I add a second type of filament and having T0 and T1 error
I see this problem when i restart firmware and upload and print. Not seeing the problem when i preheat and upload and print

**Comment by btCRO** [comment]
Had this problem for months, this pushed me away from Orca back to Prusa slicer but I want to use Orca, it is really good, fey things are missing but this one is really, reall frustrating, and it happens from version to version, and it is never corrected.

Now, I wanted to see if I'm only one, but no, so I started to play around and check gcode.
M140 command on second layer is allways missing if you change some of the settings in print configuration, not all but I found that in 100% if you change print speed (without saving and reloading profile) M140 is missing from gcode.
This is bug in program and I'm currently using V1.8.1.

Reseting change does not help, you need to change print profile to get M140 back in code.

**Reply by btCRO** [reply]
And today I found that disabling automatic brim would forgot to set M140 at second layer in 100% of situation I'we tried.

**Reply by btCRO** [reply]
I found that adding small gcode in printer settings under "Layer change G-code" solved problems for me (if slicer forget to add M140 thih adds it, and if slicer adds it it is writen twice for same layer so no problem, but this is old bug that is easy to repair...)

gcode to add is:

{if layer_num == 1}M140 S[bed_temperature]; Set print bed temperature for layers other than first {endif}

**Comment by kt0mmi** [comment]
I've also had this exact issue since day 1 of me moving to Orca Slicer (2 months now). It is pretty aggravating. It does not even happen every time but at least in 80% of my gcodes, I'd say. Please dedicate to it.

**Comment by Steven2k6** [comment]
I'm having this problem too!

Like @btCRO and @Glenn1963 said, by adding this to the "Layer Change G-Code", it is more or less fixed:
```
{if layer_num == 1}M140 S[bed_temperature]; Set print bed temperature for layers other than first {endif}
```
